### PR TITLE
Use pull request events instead of push events

### PR DESCRIPTION
**Pull request by LS80**
Using pull request synchronize events from GitHub instead of push events should fix a few issues caused by unnecessary load from pushes to long lived staging branches. With this change we only take action when a commit is pushed to a PR against `master`. We can then re-enable GitHub status updates for `ce-salesforce` and `salesforce`.

This will require turning on `Pull Requests` events in the [Webhook settings](https://github.com/organizations/FundingCircle/settings/hooks/5440962). We can also turn off `Pushes`.
image

### CODE REVIEW

File: app/controllers/github_notifications_controller.rb [hunk 1]
```diff
@@ -1,13 +1,17 @@
 # frozen_string_literal: true
 
-require 'payloads/github'
```
Comment (sgerrand) [inline app/controllers/github_notifications_controller.rb]: 💭 Did the class name need to change?

Comment (LS80) [reply app/controllers/github_notifications_controller.rb]: Not really. I just made it more explicit as it represents the payload from that specific event.

Comment (sgerrand) [reply app/controllers/github_notifications_controller.rb]: Ah, OK.

File: app/use_cases/handle_pull_request_created_event.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require 'git_repository_location'
+
+class HandlePullRequestCreatedEvent
+  include SolidUseCase
```
Comment (bliof) [inline app/use_cases/handle_pull_request_created_event.rb]: 🙋‍♂ Let's not use this gem.. The dsl is very strange and we don't really use it. Relates to #196

Comment (LS80) [reply app/use_cases/handle_pull_request_created_event.rb]: Maybe we should remove it, but I won't to do it in this PR. I'm just following the existing patterns.

File: README.md [hunk 3]
```diff
@@ -93,12 +93,19 @@ In addition to the `update_git_loop` task, you can set the environment variable
...
+a Shipment Tracker status check.
+
+To configure the webhook
+ 1. In GitHub, add a new webhook in the repository or organisation settings.
+ 2. In the API Tokens tab of Shipment Tracker, find (or create) a Github Notifications token.
+ 3. Set "Payload URL" to the GitHub notifications URL from Shipment Tracker.
+ 4. Set "Content type" to "application/json".
+ 5. Select "Let me select individual events."
+ 6. Select "Pull requests".
+ 7. Click "Add webhook"
```
Comment (sgerrand) [inline README.md]: 👏 Thank you for the documentation 💅 .

File: app/jobs/relink_ticket_job.rb [hunk 4]
```diff
@@ -11,10 +11,8 @@ def perform(args)
     before_sha = args.delete(:before_sha)
     after_sha = args.delete(:after_sha)
     full_repo_name = args.delete(:full_repo_name)
-    branch_created = args.delete(:branch_created)
 
-    if branch_created || commit_on_master?(full_repo_name, after_sha) ||
-       relink_tickets(before_sha, after_sha).empty?
+    if relink_tickets(before_sha, after_sha).empty?
```
Comment (sgerrand) [inline app/jobs/relink_ticket_job.rb]: 👏

File: app/models/payloads/github_pull_request.rb [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    def full_repo_name
+      pull_request&.dig('base', 'repo', 'full_name')
+    end
+
+    def branch_name
+      pull_request&.dig('head', 'ref')
+    end
+
+    def base_branch_master?
+      pull_request&.dig('base', 'ref') == 'master'
```
Comment (sgerrand) [inline app/models/payloads/github_pull_request.rb]: 👏

**Pull request merged**